src/flamenco/runtime/fd_runtime.c

```diff
@@ -3710,6 +3708,20 @@ void fd_update_next_epoch_stakes( fd_exec_slot_ctx_t * slot_ctx ) {
   FD_SCRATCH_SCOPE_END;
 }
 
+/* Starting a new epoch.
+  New epoch:        T
+  Just ended epoch: T-1
+  Epoch before:     T-2
+
+  In this function:
+  - leader schedule should be calculated using T-2 stakes (slot_ctx->slot_bank.epoch_stakes)
+  - stakes in T-2 (slot_ctx->slot_bank.epoch_stakes) should be replaced by T-1 (epoch_bank->next_epoch_stakes)
+  - stakes at T-1 (epoch_bank->next_epoch_stakes) should be replaced by updated stakes at T (stakes->vote_accounts)
+
+  Invariant during an epoch T:
+  epoch_bank->next_epoch_stakes    holds the stakes at T-1
+  slot_ctx->slot_bank.epoch_stakes holds the stakes at T-2
+ */
```

Comment:
So, the manifest only needs T-2 of stakes... even through they stick in 5 of them?

Reply:
would be nice to leave a comment in the code explicitly explaining this

Reply:
@jumpsiegel yes exactly, I'm not sure exactly why they store 5 as you can always derive them if you have 2 preceeding epochs